specs/RestartRequested.md

```diff
@@ -1,45 +1,89 @@
-Critical Update Available
+Restart Requested
 ===
 
 # Background
-As WebView2 developers, we often have to author ECS (experimentation and configuration service - 
-used to perform A/B testing or remotely disable features) configurations to toggle feature flags. 
-However, once these payloads are received, there is no way to restart the WebView2 and apply the
-payload. The purpose of this API is to detect such critical payloads and inform the end developer
-so they may restart their app or their WebView2 or other appropriate action.
+There are often times when WebView2 need to be restarted to apply certain update, the purpose of 
+this API is to detect whether a restart is requested from WebView2 base on different priority levels.
+WebView2 developers can listen to this event for version update, version downgrade or important
+feature flag update to determine the need to prompt user for restart to apply those updates.
 
 # Examples
-## WinRT and .NET   
+## WinRT and .NET
+### Scenario 1: Normal Priority
 ```c#
 void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
 {
     if (e.IsSuccess)
     {
         // ...
+        webView.CoreWebView2.Environment.RestartPriorityLevel = CoreWebView2RestartPriorityLevel.BestEffort;
         webView.CoreWebView2.Environment.RestartRequested += WebView_RestartRequested;
     }
 }
 
 void WebView_RestartRequested(CoreWebView2Environment sender, object e)
 {
-    // Depending on your app experience, you may or may not
-    // want to prompt user to restart the app.
+    // Depending on your app experience, you can let user
+    // to restart on normal cadence.
+    RemainderToRestartForUpdate();
+}
+```
+
+### Scenario 2: High Priority
+```c#
+void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+{
+    if (e.IsSuccess)
+    {
+        // ...
+        webView.CoreWebView2.Environment.RestartPriorityLevel = CoreWebView2RestartPriorityLevel.Critical;
+        webView.CoreWebView2.Environment.RestartRequested += WebView_RestartRequested;
+    }
+}
+
+void WebView_RestartRequested(CoreWebView2Environment sender, object e)
+{
+    // With ciritical priority, you should prompt user to save their current state and restart the app.
     RestartIfSelectedByUser();
 }
 ```
 
 ## Win32 C++
+### Scenario 1: Normal Priority
 ```cpp
 wil::com_ptr<ICoreWebView2Environment> m_webViewEnvironment;
 void CoreWebView2InitializationCompleted() {
     auto env10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
     if (env10) {
+        CHECK_FAILURE(env10->put_RestartPriorityLevel(COREWEBVIEW2_RESTART_PRIORITY_LEVEL_BEST_EFFORT));
```

Comment:
Scenario sample 1 & 2 should be combined into one event handler that checks the Priority in the event handler to decide if the update should be applied immediately or not.